dace/sdfg/dealias.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2019-2025 ETH Zurich and the DaCe authors. All rights reserved.
+import dace
+import copy
+from typing import Set, Dict
+
+from dace.sdfg.graph import MultiConnectorEdge
+from dace import SDFGState
+
+FULL_VIEW_SUFFIX = "fullview"
+SLICE_SUFFIX = "slice"
+
+
+def _get_new_connector_name(edge: MultiConnectorEdge, repldict: Dict[str, str], other_repldict: Dict[str, str],
+                            state: SDFGState, sdfg: dace.SDFG) -> str:
+    """
+    Determine new connector name for an edge based on data access patterns.
+    Following the description in the dealias routine
+
+    Args:
+        edge: The edge containing data access information
+        repldict: Dictionary of existing replacements to avoid name conflicts
+        state: The SDFG state containing the edge
+
+    Returns:
+        str: New connector name - either the original array name (for full access)
+             or a unique slice name (for partial access)
+    """
+    arr = state.sdfg.arrays[edge.data.data]
+    data_shape = arr.shape
+
+    # Full subset?
+    full_range = dace.subsets.Range([(0, dim - 1, 1) for dim in data_shape])
+    is_complete_subset = edge.data.subset == full_range
+
+    combined_repldict = repldict | other_repldict
+
+    if is_complete_subset:
+        candidate_name = edge.data.data
+        i = 1
+        while candidate_name in sdfg.arrays or candidate_name in repldict.values():
+            candidate_name = f"{edge.data.data}_{FULL_VIEW_SUFFIX}_{i}"
```

Comment:
should this not just be `{edge.data.data}_{i}`? and if so, you can use `data.find_new_name`